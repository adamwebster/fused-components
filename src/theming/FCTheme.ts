import React from 'react';

export interface Theme {
    theme?: string,
    children?: any,
}

export const FCTheme = React.createContext<Theme | null>(null);

export const FCThemeProvider = FCTheme.Provider;

export const FCThemeConsumer = FCTheme.Consumer;