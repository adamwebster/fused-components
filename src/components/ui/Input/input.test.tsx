import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import { Input } from './index';
import userEvent from '@testing-library/user-event';
import { FCThemeProvider } from '../../../theming/FCTheme';

afterEach(cleanup);

describe('FormField Input', () => {
  test('Renders the Input component', () => {
    const { getByPlaceholderText } = render(<Input id="1" placeholder="Test placeholder" />);
    expect(getByPlaceholderText('Test placeholder')).toBeInTheDocument();
  });
  test('Changes the value on change', () => {
    let initialValue = '';
    const onChange = jest.fn(e => {
      initialValue = e.target.value;
    });

    const { getByPlaceholderText, rerender } = render(
      <Input id="1" value={initialValue} onChange={e => onChange(e)} placeholder="Test placeholder" />,
    );

    const input = getByPlaceholderText('Test placeholder');
    userEvent.type(input, 'Hello World');
    expect(onChange).toBeCalledTimes(11);
    rerender(<Input id="1" value={initialValue} onChange={e => onChange(e)} placeholder="Test placeholder" />);
    expect(input).toHaveValue('Hello World');
    expect(initialValue).toBe('Hello World');
  });

  test('Renders the Input as disabled', () => {
    const { getByPlaceholderText } = render(<Input id="1" disabled placeholder="Test placeholder" />);
    expect(getByPlaceholderText('Test placeholder')).toBeDisabled();
  });

  test('Has the correct styles when in error', () => {
    const { container } = render(<Input id="1" icon="check-circle" inError placeholder="Test placeholder" />);
    expect(container).toMatchSnapshot();
  });

  test('Has the correct styles when in warning', () => {
    const { container } = render(<Input id="1" icon="check-circle" inWarning placeholder="Test placeholder" />);
    expect(container).toMatchSnapshot();
  });

  test('Has the icon when one is attached to it', () => {
    const { getByRole } = render(<Input id="1" icon="check-circle" placeholder="Test placeholder" />);
    expect(getByRole('img', { hidden: true })).toHaveClass('check-circle');
  });

  test('Is readonly', () => {
    const { getByPlaceholderText } = render(<Input id="1" readOnly placeholder="Test placeholder" />);
    expect(getByPlaceholderText('Test placeholder').getAttribute('readonly')).toBe('');
  });
  test('Sets the id', () => {
    const { getByPlaceholderText } = render(<Input id="Input control" placeholder="Test placeholder" />);
    expect(getByPlaceholderText('Test placeholder').getAttribute('id')).toBe('Input control');
  });

  test('Autocomplete can be set to off', () => {
    const { getByPlaceholderText } = render(
      <Input id="Input control" autoComplete="off" placeholder="Test placeholder" />,
    );
    expect(getByPlaceholderText('Test placeholder').getAttribute('autocomplete')).toBe('off');
  });

  test('Has the correct styles when in dark mode', () => {
    const { container } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Input id="1" icon="check-circle" placeholder="Test placeholder" />
      </FCThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
