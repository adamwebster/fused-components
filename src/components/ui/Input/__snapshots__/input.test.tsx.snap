// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField Input Has the correct styles when in dark mode 1`] = `
.c3 {
  width: 100%;
  height: 34px;
  box-sizing: border-box;
  padding: 0 10px;
  outline: none;
  border: solid 1px #627d98;
  border-radius: 5px;
  -webkit-appearance: none;
  background-color: #334e68;
  color: #DDDDDD;
  padding-left: 35px;
}

.c3::-webkit-input-placeholder {
  font-style: italic;
  color: #829ab1;
}

.c3::-moz-placeholder {
  font-style: italic;
  color: #829ab1;
}

.c3:-ms-input-placeholder {
  font-style: italic;
  color: #829ab1;
}

.c3::placeholder {
  font-style: italic;
  color: #829ab1;
}

.c3:focus {
  border-color: #627d98;
  box-shadow: 0 0 6px #627d98;
}

.c3[disabled] {
  cursor: not-allowed;
  background-color: #243b53;
}

.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  left: 10px;
  top: 7px;
  color: #627d98;
  border-right: solid 1px #627d98;
  box-sizing: border-box;
}

.c1 svg {
  top: 2px;
  position: relative;
  margin-right: 5px;
  width: 12px;
}

.c2 {
  width: 1em;
  height: 1em;
}

<div>
  <div
    class="c0"
  >
    <span
      aria-hidden="true"
      class="c1"
    >
      <svg
        class="c2  check-circle"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm-4.173,28.075l17.261,-16.676l5.559,5.754l-23.014,22.234l-0.001,-0.001l-0.001,0.001l-12.837,-13.287l5.754,-5.559l7.279,7.534Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <input
      class="c3"
      id="1"
      placeholder="Test placeholder"
    />
  </div>
</div>
`;

exports[`FormField Input Has the correct styles when in error 1`] = `
.c3 {
  width: 100%;
  height: 34px;
  box-sizing: border-box;
  padding: 0 10px;
  outline: none;
  border: solid 1px #bcbcbc;
  border-radius: 5px;
  -webkit-appearance: none;
  background-color: #fff;
  color: inherit;
  border: solid 1px #dd2c2e;
  padding-left: 35px;
}

.c3::-webkit-input-placeholder {
  font-style: italic;
  color: #999999;
}

.c3::-moz-placeholder {
  font-style: italic;
  color: #999999;
}

.c3:-ms-input-placeholder {
  font-style: italic;
  color: #999999;
}

.c3::placeholder {
  font-style: italic;
  color: #999999;
}

.c3:focus {
  border-color: #4799ff;
  box-shadow: 0 0 6px #4799ff;
}

.c3[disabled] {
  cursor: not-allowed;
  background-color: #e9e9e9;
}

.c3:focus {
  border-color: #dd2c2e;
  box-shadow: 0 0 6px #dd2c2e;
}

.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  left: 10px;
  top: 7px;
  color: #999999;
  border-right: solid 1px #bcbcbc;
  box-sizing: border-box;
  color: #dd2c2e;
  border-color: #dd2c2e;
}

.c1 svg {
  top: 2px;
  position: relative;
  margin-right: 5px;
  width: 12px;
}

.c2 {
  width: 1em;
  height: 1em;
}

<div>
  <div
    class="c0"
  >
    <span
      aria-hidden="true"
      class="c1"
    >
      <svg
        class="c2  check-circle"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm-4.173,28.075l17.261,-16.676l5.559,5.754l-23.014,22.234l-0.001,-0.001l-0.001,0.001l-12.837,-13.287l5.754,-5.559l7.279,7.534Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <input
      class="c3"
      id="1"
      placeholder="Test placeholder"
    />
  </div>
</div>
`;

exports[`FormField Input Has the correct styles when in warning 1`] = `
.c3 {
  width: 100%;
  height: 34px;
  box-sizing: border-box;
  padding: 0 10px;
  outline: none;
  border: solid 1px #bcbcbc;
  border-radius: 5px;
  -webkit-appearance: none;
  background-color: #fff;
  color: inherit;
  border: solid 1px #eba300;
  padding-left: 35px;
}

.c3::-webkit-input-placeholder {
  font-style: italic;
  color: #999999;
}

.c3::-moz-placeholder {
  font-style: italic;
  color: #999999;
}

.c3:-ms-input-placeholder {
  font-style: italic;
  color: #999999;
}

.c3::placeholder {
  font-style: italic;
  color: #999999;
}

.c3:focus {
  border-color: #4799ff;
  box-shadow: 0 0 6px #4799ff;
}

.c3[disabled] {
  cursor: not-allowed;
  background-color: #e9e9e9;
}

.c3:focus {
  border-color: #eba300;
  box-shadow: 0 0 6px #eba300;
}

.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  left: 10px;
  top: 7px;
  color: #999999;
  border-right: solid 1px #bcbcbc;
  box-sizing: border-box;
  color: #eba300;
  border-color: #eba300;
}

.c1 svg {
  top: 2px;
  position: relative;
  margin-right: 5px;
  width: 12px;
}

.c2 {
  width: 1em;
  height: 1em;
}

<div>
  <div
    class="c0"
  >
    <span
      aria-hidden="true"
      class="c1"
    >
      <svg
        class="c2  check-circle"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24,0c13.246,0 24,10.754 24,24c0,13.246 -10.754,24 -24,24c-13.246,0 -24,-10.754 -24,-24c0,-13.246 10.754,-24 24,-24Zm-4.173,28.075l17.261,-16.676l5.559,5.754l-23.014,22.234l-0.001,-0.001l-0.001,0.001l-12.837,-13.287l5.754,-5.559l7.279,7.534Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <input
      class="c3"
      id="1"
      placeholder="Test placeholder"
    />
  </div>
</div>
`;
